presto-main/src/main/java/com/facebook/presto/operator/scalar/sql/ArraySqlFunctions.java

```diff
@@ -91,6 +91,26 @@ public static String arrayHasDuplicatesVarchar()
         return "RETURN cardinality(array_duplicates(input)) > 0";
     }
 
+    @SqlInvokedScalarFunction(value = "array_least_frequent", deterministic = true, calledOnNullInput = true)
+    @Description("Determines the least frequent elemement in the array. If there are multiple elements, the function returns the biggest element")
+    @TypeParameter("T")
+    @SqlParameter(name = "input", type = "array(T)")
+    @SqlType("array<T>")
+    public static String array_least_frequent()
+    {
+        return "RETURN IF(COALESCE(CARDINALITY(REMOVE_NULLS(input)), 0) = 0, NULL, TRANSFORM(SLICE(ARRAY_SORT(MAP_ENTRIES(ARRAY_FREQUENCY(REMOVE_NULLS(input))),(a, b) -> IF(a[2] > b[2], 1, IF(a[2] = b[2], IF(a[1] < b[1], 1, -1), -1))), 1, 1), x -> x[1]))";
+    }
+
+    @SqlInvokedScalarFunction(value = "array_least_frequent", deterministic = true, calledOnNullInput = true)
+    @Description("Determines the n least frequent elemement in the array in the descending order of the elements.")
+    @TypeParameter("T")
+    @SqlParameters({@SqlParameter(name = "input", type = "array(T)"), @SqlParameter(name = "n", type = "bigint")})
+    @SqlType("array<T>")
+    public static String array_n_least_frequent()
```

Comment:
To be consistent with others like MAP_TOP_N etc, just make _N a suffix (we should also see if we can overload the function later).